packages/dds/tree/src/simple-tree/types.ts

```diff
@@ -67,4 +83,140 @@ export abstract class TreeNode implements WithType {
 	 * Subclasses provide more specific strings for this to get strong typing of otherwise type compatible nodes.
 	 */
 	public abstract get [type](): string;
+
+	protected constructor() {
+		if (!(this instanceof TreeNodeValid)) {
+			throw new UsageError("TreeNodes must extend schema classes created by SchemaFactory");
+		}
+	}
+}
+
+/**
+ * Class which all {@link TreeNode}s must extend.
+ * Since this is not exported, it allows robust detection of attempts to create TreeNodes which do not go through SchemaFactory which is the only place which exposes classes that extend this.
+ *
+ * This has static members which schema classes can override to provide schema specific functionality.
+ * These static members are only intended to be used / overridden by code within this package, and are used by the various node kinds.
+ * Access to these static members has to be done via `this.constructor.staticMember` to support the overrides, and thus can only be used in the constructor, after the base constructor has been invoked.
+ */
+export abstract class TreeNodeValid<TInput> extends TreeNode {
+	/**
+	 * Schema classes can override this to control what happens at the end of the constructor.
+	 * The return value from this is returned from the constructor, allowing substituting a proxy if desired.
+	 *
+	 * This is not simply done in the derived constructor to enable:
+	 * - this class to access the value which is being returned before its returned from the constructor.
+	 * - the derived class to be provided the input `FlexTreeNode` without relying on a field on the node to hold it.
+	 */
+	protected static prepareInstance<T>(
+		this: typeof TreeNodeValid<T>,
+		instance: TreeNodeValid<T>,
+		input: FlexTreeNode,
+	): TreeNodeValid<T> {
+		return instance;
+	}
+
+	/**
+	 * Schema classes must override to provide an implementation of RawTreeNode construction.
+	 */
+	protected static buildRawNode<T>(
+		this: typeof TreeNodeValid<T>,
+		instance: TreeNodeValid<T>,
+		input: T,
+	): RawTreeNode<FlexTreeNodeSchema, unknown> {
+		return fail("Schema must override buildRawNode");
+	}
+
+	/**
+	 * Schema classes can override to provide a callback that once when the first node is constructed.
+	 * This is a good place to perform extra validation and cache schema derived data needed for the implementation of the node.
+	 */
+	protected static oneTimeSetup<T>(this: typeof TreeNodeValid<T>) {}
+
+	/**
+	 * Used to ensure we only use one most derived schema type:
```

Comment:
wording here is a bit confusing

Reply:
I have rewritten this, maybe its better now.